log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -141,6 +147,25 @@ public Snapshot createSnapshot() {
         }
     }
 
+    private void decompressAndCopyGZipFile() throws IOException {
+
+        File tempFile = File.createTempFile("log-viewer-", "-" + file.getName(file.getNameCount() - 1) + ".tmp");
+        tempFile.deleteOnExit();
+
+        try (GZIPInputStream gis = new GZIPInputStream(
+                Files.newInputStream(file.toFile().toPath()));
```

Comment:
Is it really necessary to convert it to File, and then back to Path? Or is it just mistake?